src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -13,8 +13,9 @@ namespace Microsoft.CodeAnalysis.CSharp
     internal class ExpressionVariableFinder : CSharpSyntaxWalker
     {
         private Binder _scopeBinder;
-        private Binder _enclosingBinderOpt;
+        private Binder _enclosingBinder;
         private ArrayBuilder<LocalSymbol> _localsBuilder;
+        private SyntaxNode _nodeToBind;
```

Comment:
Yes. Also we also will have the correct binder and don't need to look it up in `DeconstructionLocalSymbol.InferTypeOfVarVariable`.

---
In reply to: [76325534](https://github.com/dotnet/roslyn/pull/13274#discussion_r76325534) [](ancestors = 76325534)